### DISCUSSION TITLE
Adding a trunk to connect to VTO sip server - Need sip_custom_general.conf and extensions_custom.conf configurations files

### DISCUSSION THREAD

**Discussion by naftalih** [TECH7Fox/asterisk-hass-addons#317] [Q&A]
Hi,
I have, for many years now, a setup of VTO intercoms and VTH monitors in conjunction with FreePBX. The logic is that Dahua devices works best between themeselves when one of the VTOs is acting as a SIP server (most reliable). So my FreePBX is connectiong to the VTO SIP server as an extension (a trunk). Now that I added the asterisk add on to HA, I dont need 2 asterisk servers on my network. So I would like to setup the HA asterisk add on as the main server for the intercoms. In order to do that I need the option to add the trunk registration (register=>) to the [general] section at sip.conf - on FreePBX you have a file called sip_general_custom.conf which is included in sip.conf under [general] section.
In addition to that I need to create my own dial plan for calls from the VTOs (intercoms) and a RIng Group at extensions.conf.

The question is, how do I customize the [general] section of sip.conf and how to customize extensions.conf?

Thanks a lot.

**Comment by felipecrs** [comment]
If I understood correctly: just copy `/config/asterisk/default/sip.conf` to `/config/asterisk/custom/sip.conf` and make your changes, this is in the add-on documentation.

**Comment by naftalih** [comment]
I tried it. It seems to work only with "custom" conf files

**Reply by felipecrs** [reply]
Like which ones?

Also, are you running as an add-on or as a normal docker container? If the latter, perhaps you are affected by https://github.com/TECH7Fox/asterisk-hass-addons/issues/309#issuecomment-1961078570?

**Comment by naftalih** [comment]
Run as add-on

**Comment by naftalih** [comment]
sip_custom.conf and pjsip_custom.conf works but I need custom extensions and to customize the [general] part of sip.conf

**Reply by felipecrs** [reply]
Take a look here, are you doing like this? https://github.com/TECH7Fox/asterisk-hass-addons/issues/309#issuecomment-1960132654

**Comment by naftalih** [comment]
Thanks for your help!
Just to make sure I understand correctly:
1. When creating a file in custom folder which is not custom (like
sip_custom.conf which is #included in sip.conf) then I need the full
content of the original file + my additions? For example, extensions.conf
in custom folder should include all content of original extensions.conf in
default folder + my dial plans...(which is what I did...)
2. Do I need to do a full server restart or add-on restart is enough for
those changes to take place?

Thanks again.

**Reply by felipecrs** [reply]
1. Yes, and this is valid for all files.
2. Add-on restart.

**Comment by naftalih** [comment]
Thanks. That's what I did but will give it a try again. Maybe I
did something wrong.

**Comment by naftalih** [comment]
Thanks. It works great.
